sql/catalyst/src/main/scala/org/apache/spark/sql/types/StructType.scala

```diff
@@ -446,7 +446,13 @@ case class StructType(fields: Array[StructField]) extends DataType with Seq[Stru
   }
 
   override def sql: String = {
-    val fieldTypes = fields.map(f => s"${quoteIdentifier(f.name)}: ${f.dataType.sql}")
+    val fieldTypes = fields.map(f => {
+      val comment = f.getComment()
+        .map(escapeSingleQuotedString)
+        .map(" COMMENT '" + _ + "'")
```

Comment:
Maybe move `getOrElse("")` here, then next line will look simpler `s"${quoteIdentifier(f.name)}: ${f.dataType.sql}$comment`.

Reply:
Thank you! That was bugging me as well. 

I feel like I am duplicating code from [here](https://github.com/apache/spark/blob/master/sql/catalyst/src/main/scala/org/apache/spark/sql/types/StructField.scala#L92) but I couldn't think of an elegant way to do it. Any suggestions?

Reply:
Maybe I can add `def getSQLComment: String` under `StructField`?

Reply:
maybe we can add `StructField.sql`, and here can be `s"STRUCT<${fields.map(_.sql).mkString(", ")}>"`, so that it's consistent with DDL.

Reply:
I think I found a solution. less code == better right?